js/data.js

```diff
@@ -13,6 +14,11 @@ if (previousEntries) data = JSON.parse(previousEntries);
 window.addEventListener('beforeunload', handleUnload);
 
 function handleUnload(event) {
+  var currentView = '';
+  for (var i = 0; i < $viewList.length; i++) {
+    if (!$viewList[i].classList.contains('hidden')) currentView = $viewList[i].getAttribute('data-view');
+  }
+  data.view = currentView;
```

Comment:
If you update the value of `data.view` every time the view is changed, you can avoid needing to do this loop before the page unloads.